### DISCUSSION TITLE
Upgrading v4 to v6

### DISCUSSION THREAD

**Discussion by fjuriolli** [react-hook-form/react-hook-form#4137] [Q&A]
Heya fellas, I'm still using the `"react-hook-form": "^4.2.2"` version.

In the project, basically, all of the input is using the `Controller`, like:

```javascript
    <Controller
        as={<TextField fullWidth variant="outlined" placeholder="Type..." />}
        rules={{ required: t('defaultValidation', { inputName: t('title') }) }}
        name="title"
        control={control}
     />
    <ErrorMessage errors={errors} name="title" as="p" />
```
and:

```javascript
    <Controller
      as={
        <Select
          placeholderText="Select..."
          options={selectOptions}
        />
      }
      name="campaigns"
      control={control}
      onChange={([selected]) => {
        return selected
      }}
    />
```

I was checking the changelogs and I saw a couple of changes, such as the new `render` prop.

Any tips to a smooth migration? And yeah I'm using v4 but should I follow the `v5 to v6 migration guide`?

**Comment by bluebill1049** [comment]
you can follow the migration guide yes, v4 and v5 pretty much the same. please migrate to use `render` prop, it's more future proof.

**Comment by Leelsmuth** [comment]
Hi @bluebill1049 , I am still using the "react-hook-form": "^4.10.2" version.

I am trying to update to version 7 but basically it's breaking my code.

Alternatively, I have tried to update incrementally from v4 to v5, v5 to v6 and then to v7 but the migration guide for v4 to v5 and v5 to v6 return 404s

https://react-hook-form.com/migration-v6#changes-from-v5-to-v6
https://react-hook-form.com/migration-v5#changes-from-v4-to-v5

**Comment by leapful** [comment]
@Leelsmuth 

I think there is no migration guide for v4 to v5 as it does not introduce many breaking changes. You can check the comment of Bill above to start upgrading react-hook-form to v5, just start using `render` prop for `Controller` component.

For migration guide from v5 to v6, you can find here https://legacy.react-hook-form.com/migrate-v5-to-v6/.
